src/polyglot-notebooks-vscode-common/src/vscodeNotebookManagement.ts

```diff
@@ -31,13 +31,44 @@ export async function replaceNotebookCellMetadata(notebookUri: vscode.Uri, cellI
 }
 
 export async function replaceNotebookMetadata(notebookUri: vscode.Uri, documentMetadata: { [key: string]: any }): Promise<boolean> {
-    const notebookEdit = vscode.NotebookEdit.updateNotebookMetadata(documentMetadata);
-    const edit = new vscode.WorkspaceEdit();
-    edit.set(notebookUri, [notebookEdit]);
-    const succeeded = await vscode.workspace.applyEdit(edit);
-    return succeeded;
+    const notebook = vscode.workspace.notebookDocuments.find(d => d.uri === notebookUri);
+    if (notebook) {
+        const metadata = notebook.metadata;
+        const shouldUpdate = compare(metadata, documentMetadata);
+        if (shouldUpdate) {
+            const notebookEdit = vscode.NotebookEdit.updateNotebookMetadata(documentMetadata);
+            const edit = new vscode.WorkspaceEdit();
+            edit.set(notebookUri, [notebookEdit]);
+            const succeeded = await vscode.workspace.applyEdit(edit);
+            return succeeded;
+        }
+        return false;
```

Comment:
nit: This line seems redundant given the return below
```suggestion
```

Reply:
got it